### DISCUSSION TITLE
Unpoly 2.1.0 released

### DISCUSSION THREAD

**Discussion by triskweline** [unpoly/unpoly#230] [Announcements]
### Built-in loading indicator

Unpoly now shows an animated [progress bar](https://unpoly.com/up.network.config#config.progressBar) that appears for [late requests](https://unpoly.com/up:request:late).

![Video of progress bar animation](https://user-images.githubusercontent.com/89953/125434912-bd54c75d-712a-47f6-94c9-e75efa6fb8b1.gif)

The progress bar is enabled by default. If you're using [`unpoly-migrate.js`](https://unpoly.com/changes/upgrading), the progress bar is disabled if you have an `up:request:late` listener, assuming that you have built a custom loading indicator.

### Simplified history options for new overlays

For new layers, the `[up-history-visible]` and `[up-history]` options have been unified into a single `[up-history]` option. This reverts to the old behavior of Unpoly 1.0. The separation into two options was introduced in Unpoly 2.0, but turned out to be confusing to users.

### Quality of life improvements

- [Layer configuration](https://unpoly.com/up.layer.config) may now set mode-specific defaults for [`{ scroll }`](https://unpoly.com/scroll-option) and [`{ focus }`](https://unpoly.com/focus-option). These take precendence to defaults in [`up.fragment.config.navigateOptions`](https://unpoly.com/up.fragment.config#config.navigateOptions).
- Links with an [`[up-instant]`](https://unpoly.com/a-up-instant) attribute are now followed automatically, even if they don't also have an [`[up-follow]`](https://unpoly.com/a-up-follow) attribute.

**Comment by ivangretsky** [comment]
Thanks for the great work!
